pkg/pillar/devicenetwork/dnc.go

```diff
@@ -463,7 +486,7 @@ func getNextTestableDPCIndex(ctx *DeviceNetworkContext, start int) int {
 		newIndex = (newIndex + 1) % dpcListLen
 	}
 	if count == dpcListLen {
-		newIndex = 0
+		newIndex = -1
```

Comment:
This means that we have not found a working DPC from the entire DPC list, am I right?. Should we do something drastic like rebooting the device. Peripheral device errors might some times drive us into this situation. A reboot might help.

Reply:
This actually what we get when the server cert has expired. Prior to the fix it would loop forever trying index 0 until a watchdog due to nim.touch.

With the changes we sit cool with the IP addresses in place, and 5 minutes later or so try again.